lib/routes/podcast/custom_amount_form.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:breez/bloc/user_profile/currency.dart';
+import 'package:breez/widgets/sat_amount_form_field_formatter.dart';
+import 'package:fixnum/fixnum.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class CustomAmountForm extends StatelessWidget {
+  final GlobalKey<FormState> formKey;
+  final FocusNode focusNode;
+  final TextEditingController textEditingController;
+  final List<int> preset;
+  final InputDecoration decoration;
+
+  const CustomAmountForm({
+    Key key,
+    this.formKey,
+    this.focusNode,
+    this.textEditingController,
+    this.preset: const [1],
+    this.decoration,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    return Form(
+      key: formKey,
+      autovalidateMode: AutovalidateMode.disabled,
+      child: TextFormField(
+        autovalidateMode: AutovalidateMode.disabled,
+        focusNode: focusNode,
+        controller: textEditingController,
+        keyboardType: TextInputType.number,
+        inputFormatters: [
+          SatAmountFormFieldFormatter(),
+        ],
+        validator: (raw) {
+          if (raw.length == 0) {
+            return "Please enter a custom amount";
+          }
+          int value = Currency.SAT.parseToInt(raw);
+          if (value < preset[0]) {
+            return "Please enter at least ${preset[0]} sats.";
+          }
+          return null;
+        },
+        style: theme.dialogTheme.contentTextStyle.copyWith(height: 1.0),
+        decoration: decoration,
+      ),
+    );
+  }
+}
+
+String initialCustomAmount(
```

Comment:
For this component to be more self contained maybe it is better to initialize the textEditingController here instead of calling this method from outside?
We can pass the presetAmounts and customAmount to the widget.

Reply:
@roeierez I did try it but it stops the field to work, I'm still trying to understand the reason under the hoods of this but my guess, for now, is that it messes with form<->controller relationship.

I thought of an alternative way and I would like to know what you think about it, the alternative is to have a custom `TextEditingController` let's say `CustomAmountTextEditingController` and pass it for `CustomAmountForm` instead of a normal `TextEditingController`. that way we set the initial text logic inside the `CustomAmountTextEditingController`, what do you think?